src/analysis/rust_type.rs

```diff
@@ -79,11 +79,15 @@ fn rust_type_full(env: &Env, type_id: library::TypeId, nullable: Nullable, by_re
         rust_type = rust_type.map_any(|s| format!("&{}", s));
     }
     if *nullable {
-        // ignore nullability of containers, they don't require wrapping in an Option
-        match TypeKind::of(&env.library, type_id) {
-            TypeKind::Pointer |
-                TypeKind::Object |
-                TypeKind::Interface => rust_type = rust_type.map_any(|s| format!("Option<{}>", s)),
+        use analysis::conversion_type::ConversionType::*;
+        match ConversionType::of(&env.library, type_id) {
+            Owned => {
+                // ignore nullability of containers, they don't require wrapping in an Option
+                match *type_ {
+                    library::Type::List(..) => (),
```

Comment:
I've probably forgotten something but now it seems the conversion type doesn't matter here: it's sufficient to check if this is a container (only `List` here because I didn't feel like enumerating all containers).